flight/Modules/VtolPathFollower/vtol_follower_control.c

```diff
@@ -627,6 +529,142 @@ int32_t vtol_follower_control_attitude(float dT)
 	return 0;
 }
 
+
+static float loiter_deadband(float input) {
+	const float CMD_THRESHOLD = 0.2f;
+
+	if (input > CMD_THRESHOLD) {
+		input -= CMD_THRESHOLD;
+	} else if (input < -CMD_THRESHOLD) {
+		input += CMD_THRESHOLD;
+	} else {
+		input = 0;
+	}
+
+	input /= (1 - CMD_THRESHOLD);	// Normalize to -1 to 1 range.
+
+	/* Confine to desired range */
+	if (input > 1.0f) {
+		input = 1.0f;
+	} else if (input < -1.0f) {
+		input = -1.0f;
+	}
+
+	return expo3(input, 40);	// And apply 40% expo
+}
+
+/**
+ * Receives loiter commands and makes appropriate adjustments.
+ * @param[in] dT the time since last evaluation
+ * @param[in,out] hold_pos the hold position
+ * @param[out] att_adj an adjustment to be made to attitude for responsiveness.
+ */
+
+bool vtol_follower_control_loiter(float dT, float *hold_pos, float *att_adj) {
+	LoiterCommandData cmd;
+	LoiterCommandGet(&cmd);
+
+	// XXX TODO reproject when we're not issuing body-centric commands
+	float commands_rp[2] = {
+		cmd.Roll,
+		cmd.Pitch
+	};
+
+	float command_mag = vectorn_magnitude(commands_rp, 2);
+	float deadband_mag = loiter_deadband(command_mag);
+
+	static float historic_mag = 0.0f;
```

Comment:
normally i don't ask for comments, but could you document this structure. 

there a peak hold detection and decay going on, and we only do things below when there was a historical stick magnitude that falls within a certain historical envelope. i'm guessing because there is further processing to do after the stick goes back to neutral for some time window...